svc-update/update/getresources.go

```diff
@@ -87,3 +95,161 @@ func GetUpdateService() response.RPC {
+		}
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
+	}
+	var resource map[string]interface{}
+	json.Unmarshal([]byte(data), &resource)
+	resp.Body = resource
+	resp.StatusCode = http.StatusOK
+	resp.StatusMessage = response.Success
+
+	return resp
+
+}
+
+// GetAllSoftwareInventory is a functioanlity to retrive all the available inventory
+// resources from the added BMC's
+func (e *ExternalInterface) GetAllSoftwareInventory(req *updateproto.UpdateRequest) response.RPC {
+	var resp response.RPC
+	resp.Header = map[string]string{
+		"Allow":             `"GET"`,
+		"Cache-Control":     "no-cache",
+		"Connection":        "keep-alive",
+		"Content-type":      "application/json; charset=utf-8",
+		"Transfer-Encoding": "chunked",
+		"OData-Version":     "4.0",
+	}
+	softwareCollection := uresponse.Collection{
+		OdataContext: "/redfish/v1/$metadata#SoftwareInventoryCollection.SoftwareCollection",
+		OdataID:      "/redfish/v1/UpdateService/SoftwareInventory",
+		OdataType:    "#SoftwareInventoryCollection.SoftwareInventoryCollection",
+		Description:  "SoftwareInventory view",
+		Name:         "SoftwareInventory",
+	}
+	var members []dmtf.Link
+
+	softwareCollectionKeysArray, err := e.DB.GetAllKeysFromTable("SoftwareInventory")
+	if err != nil || len(softwareCollectionKeysArray) == 0 {
+		log.Printf("odimra Doesnt have Servers")
+	}
+
+	for _, key := range softwareCollectionKeysArray {
+		members = append(members, dmtf.Link{Oid: key})
+	}
+	softwareCollection.Members = members
+	softwareCollection.MembersCount = len(members)
+	resp.Body = softwareCollection
+	resp.StatusCode = http.StatusOK
+	return resp
+}
+
+// GetSoftwareInventory is used to fetch resource data. The function is supposed to be used as part of RPC
+// For getting software inventory resource information,  parameters need to be passed Request .
+// Request holds the  Uuid and Url ,
+// Url will be parsed from that search key will created
+// There will be two return values for the fuction. One is the RPC response, which contains the
+// status code, status message, headers and body and the second value is error.
+func (e *ExternalInterface) GetSoftwareInventory(req *updateproto.UpdateRequest) response.RPC {
+	var resp response.RPC
+	resp.Header = map[string]string{
+		"Allow":             `"GET"`,
+		"Cache-Control":     "no-cache",
+		"Connection":        "keep-alive",
+		"Content-type":      "application/json; charset=utf-8",
+		"Transfer-Encoding": "chunked",
+		"OData-Version":     "4.0",
+	}
+
+	requestData := strings.Split(req.ResourceID, ":")
+	if len(requestData) <= 1 {
+		errorMessage := "error: SystemUUID not found"
+		return common.GeneralError(http.StatusNotFound, response.ResourceNotFound, errorMessage, []interface{}{"SoftwareInventory", req.ResourceID}, nil)
+	}
+	data, gerr := e.DB.GetResource("SoftwareInventory", req.URL)
```

Comment:
While getting the System, Chassis or Manager Resource, if it is not found in DB the  resource, will be fetched from the device and stored again please confirm once whether that logic needs to be added here

Reply:
addressed